community/common/src/main/java/org/neo4j/kernel/api/exceptions/Status.java

```diff
@@ -422,6 +422,8 @@ public Code code()
                 "A malformed schema rule was encountered. Please contact your support representative." ),
         UnknownError( DatabaseError,
                 "An unknown error occurred." ),
+        OutOfMemoryError( TransientError,
+                "There is not enough memory to perform the current task." ),
```

Comment:
OOMs might leave the database partially crashed, though, and require a restart of the JVM. They can break locking protocols and arbitrary database services, so recommending a restart would be the safe approach.